packages/dds/tree/src/feature-libraries/editable-tree/editableTreeSequence.ts

```diff
@@ -0,0 +1,335 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert } from "@fluidframework/common-utils";
+import { ITreeSubscriptionCursor, TreeNavigationResult } from "../../forest";
+import { NamedTreeSchema, TreeSchemaIdentifier, lookupTreeSchema } from "../../schema-stored";
+import { Anchor, FieldKey } from "../../tree";
+import {
+    EditableTree,
+    EditableTreeOrPrimitive,
+    getTypeSymbol,
+    inProxyOrUnwrap,
+    ProxyTarget,
+    proxyTargetSymbol,
+    UnwrappedEditableField,
+} from "./editableTree";
+import { ProxyContext } from "./editableTreeContext";
+import { AdaptingProxyHandler, getArrayOwnKeys, keyIsValidIndex } from "./utilities";
+
+/**
+ * Sequence field nodes as a sequence of EditableTrees or primitive values (if unwrapped).
+ */
+export interface EditableTreeSequence<T extends EditableTreeOrPrimitive = EditableTreeOrPrimitive>
```

Comment:
One way I think we could simplify the users of this type, is to remove the generic type parameter. 

Instead of getting a EditableTreeSequence<EditiableTree> or a EditableTreeSequence<UnwrappedEditiableTree>, we can have a single sequence type, which will give you UnwrappedEditiableTrees by default, but we can add an API to get an EditiableTree explicitly (ex: `getWithoutUnwraping(index: number): EditiableTree`).